lib/strategy/consul/agent.ex

```diff
@@ -7,10 +7,42 @@ defmodule Cluster.Strategy.Consul.Agent do
   use Cluster.Strategy.Consul.Endpoint
 
   @impl true
-  def build_url(%URI{} = url, config) do
-    %{url | path: "/agent/service/#{config[:service_name]}"}
+  def build_url(%URI{query: query} = url, config) do
```

Comment:
This is a breaking change.

Before users were used to add `service_name`, and now they will need to use `:id` or `:service` instead. Very important to state this very clearly at the `README.md` and even instruct users how to migrate their code to the new version.

Reply:
I have pushed changes to Readme file.